### ISSUE TITLE
Rule proposal: No moot try/catch

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#566]
I sometimes see code like this:

```js
try {
	const result = await pMap(names, name => request(name, options));
	return new Map(zip(names, result));
} catch (error) {
	throw error;
}
```

In this case, the try/catch is moot as the error is just directly rethrown.

So the rule would trigger if the only thing in the catch block is `throw errorVariableName;`.

**Comment by silverwind** [comment]
FWIW, this is already implemented in rule [sonarjs/no-useless-catch](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-useless-catch.md).

**Comment by sindresorhus** [comment]
@silverwind It would still be useful here though. I’m not going to depend on the SonarJS plugin just for one rule. It’s also a very simple rule to make.

**Comment by sindresorhus** [comment]
Oh, interesting. Seems it didn't catch https://github.com/sindresorhus/npm-name/pull/38#discussion_r382889830 as it was not enabled yet then: https://github.com/xojs/eslint-config-xo/commit/28608a4f28b2eb9f0ccccd8f6daecf97bc7f633d
